repos/system_upgrade/common/actors/scansourcefiles/tests/unit_test_scansourcefiles.py

```diff
@@ -1,5 +1,130 @@
-def test_scansourcefiles():
-    # TODO(pstodulk): keeping unit tests for later after I check the idea
-    # of this actor with the team.
-    # JIRA: OAMG-10367
-    pass
+import os
+
+import pytest
+
+from leapp.libraries.actor import scansourcefiles
+from leapp.libraries.common import testutils
+from leapp.libraries.stdlib import api
+from leapp.models import FileInfo, TrackedFilesInfoSource
+
+CUR_DIR = os.path.dirname(os.path.abspath(__file__))
+FILES_DIR = os.path.join(CUR_DIR, 'files')
+
+# values in list should correspond with the return values of get_source_major_version()
+RHEL_MAJOR_VERSIONS_LIST = list(range(8, 9+1))
+
+TRACKED_FILES_BY_VERSIONS = {
+    version: ['{}/file_major_version_{}'.format(FILES_DIR, version)]
+    for version in RHEL_MAJOR_VERSIONS_LIST
+    }
+
+VERSION_FILES_MOCKED = {
+    'common': [
+        '{}/file_not_rpm_owned'.format(FILES_DIR),
+    ]
+}
+
+VERSION_FILES_MOCKED.update(TRACKED_FILES_BY_VERSIONS)
+
+
+@pytest.mark.parametrize('major_version', RHEL_MAJOR_VERSIONS_LIST)
+def test_version_files_with_common(monkeypatch, major_version):
+    monkeypatch.setattr(api, 'produce', testutils.produce_mocked())
+    monkeypatch.setattr(api, 'current_logger', testutils.logger_mocked())
+    monkeypatch.setattr(scansourcefiles, 'get_source_major_version', lambda: major_version)
+    monkeypatch.setattr(scansourcefiles, 'TRACKED_FILES', VERSION_FILES_MOCKED)
+    scansourcefiles.process()
+
+    assert api.produce.called == 1
+    assert isinstance(api.produce.model_instances[0], TrackedFilesInfoSource)
+    # assert only files from correct major version were scanned, so common files + 1 (version file)
+    assert len(api.produce.model_instances[0].files) == len(VERSION_FILES_MOCKED['common']) + 1
+
+    assert isinstance(api.produce.model_instances[0].files[0], FileInfo)
+    assert api.produce.model_instances[0].files[0].path == '{}/file_not_rpm_owned'.format(FILES_DIR)
+    assert api.produce.model_instances[0].files[0].exists is True
+    assert api.produce.model_instances[0].files[0].rpm_name == ''
+    assert api.produce.model_instances[0].files[0].is_modified is False
+
+    INDEX_OF_VERSION_FILE = len(VERSION_FILES_MOCKED['common'])
```

Comment:
You could store `api.produce.model_instances[0].files[INDEX_OF_VERSION_FILE]` inside a variable instread to avoid a lot of the repetition below.

Reply:
Was hesitant about this, because didn't see other actors doing this (didn't check enough of them). But yes, there is no reason for this repetition, which makes it worse

Reply:
It isn't really an issue, but since you already have variable for the index you might as well store the object since you don't need the index anywhere else.